src/mwutils/mwmath.cpp

```diff
@@ -18,12 +17,10 @@ using namespace Eigen;
 using namespace mrcpp;
 
 /** Calculate \f$ m^e\f$ for integers. */
-int MathUtils::ipow(int m, int e) {
-    if (e < 0) {
-        MSG_FATAL("Exponent cannot be negative: " << e)
-    }
+int mwmath::ipow(int m, int e) {
```

Comment:
You could probaly use `size_t` or `unsigned int` to have the compiler check for you that an positive integer was passed. Also a `static_assert` rather than a runtime check might be better here. Out of curiosity, does this outperform `std::pow`?

Reply:
I guess the purpose of this `ipow` is that it returns an `int`, all versions of `std::pow` returns `floats`, right?

Reply:
Yes, http://en.cppreference.com/w/cpp/numeric/math/pow
Out of curiosity, where is this used?

Reply:
It's mostly used to compute `(k+1)^D`, where `1 < k < 20` is the polynomial order, and `1 < D < 3` is the dimension. Could easily be replaced by inline evaluations like
```
int result = (k+1)*(k+1);
```
or
```
int result = 1;
for (int d = 0; d < D; d++) result *= (k+1);
```

Reply:
But in the end this `ipow`is not performance critical, it's just for convenience.

Reply:
Yeah, sorry. It just piqued my interest :)